### ISSUE TITLE
argo-server manifest missing `--secure` flag cause ReadinessProbe handshake failed

### ISSUE THREAD

**Issue by book987** [argoproj/argo-workflows#5412]
## Summary 

According to #5228 , argo-server deployment enable HTTPS probe by default, but does not specify `--secure` options in args. This cause readiness probe failed with `tls: first record does not look like a TLS handshake`, deployment argo-server keeps unhealthy.

## Diagnostics

What version of Argo Workflows are you running? 
v3.0.0-rc6

```
Readiness probe failed: Get https://xx.xxx.xxx.xxx:2746/: tls: first record does not look like a TLS handshake
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

**Comment by book987** [comment]
already fix in #5212 , but not in v3.0.0-rc6, will this pr be picked in next rc? @alexec

**Comment by alexec** [comment]
#5409
